### PR TITLE
Add ZenML as official Docker template

### PR DESCRIPTION
**Pull request by strickvl**
Making this PR ahead of becoming an official template. (CC: @davanstrien)

I wasn't sure if the `{% hint }` markdown syntax was supported. Happy to amend if there's a better way to do those. I see that `<Tip></Tip>` is used in some places, but I couldn't figure out the equivalent for some kind of a warning.

The photo used to illustrate using the official template will need updating as / when the template gets set up.

### CODE REVIEW

**Review by davanstrien — COMMENTED** [review]
@davanstrien update this once we have an updated screenshot

File: docs/assets/zenml/choose_space.png
Comment (davanstrien) [inline docs/assets/zenml/choose_space.png:1]: @davanstrien update this

File: docs/assets/zenml/hf_spaces_chart.png
Comment (davanstrien) [inline docs/assets/zenml/hf_spaces_chart.png:1]: @osanseviero do we prefer all doc images to go in https://huggingface.co/datasets/huggingface/documentation-images?

File: docs/hub/spaces-sdks-docker-zenml.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
+# ZenML on Spaces
+
+[ZenML](https://github.com/zenml-io/zenml) is an extensible, open-source MLOps framework for creating portable, production-ready MLOps pipelines. It's built for Data Scientists, ML Engineers, and MLOps Developers to collaborate as they develop to production.
+
+ZenML offers a simple and flexible syntax, is cloud- and tool-agnostic, and has
+interfaces/abstractions catered toward ML workflows. With ZenML you'll have all
+your favorite tools in one place, so you can tailor a workflow that caters to
+your specific needs.
+
+This Huggingface Space allows you to get up and running with a deployed version
```
Comment (davanstrien) [inline docs/hub/spaces-sdks-docker-zenml.md:10]:
```suggestion
The ZenML Huggingface Space allows you to get up and running with a deployed version
```

File: docs/hub/spaces-sdks-docker-zenml.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+You can also use the Direct URL in your browser to use the ZenML dashboard as a
+fullscreen application (i.e. without the HuggingFace Spaces wrapper around it).
+
+## Extra Configuration Options
+
+By default the ZenML application will be configured to use a SQLite
+non-persistent database. If you want to use a persistent database, you can
+configure this by amending the `Dockerfile` in your Space's root directory. For
+full details on the various parameters you can change, see [our reference
+documentation](./docker.md#zenml-server-configuration-options) on configuring
```
Comment (davanstrien) [inline docs/hub/spaces-sdks-docker-zenml.md:102]: Should this be pointing to https://docs.zenml.io/getting-started/deploying-zenml/docker#zenml-server-configuration-options?

File: docs/hub/spaces-sdks-docker-zenml.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+ZenML when deployed with Docker.
+
+{% hint style="info" %}
+If you are using the space just for testing and experimentation, you don't need
+to make any changes to the configuration. Everything will work out of the box.
+{% endhint %}
+
+You can also use an external secrets backend together with your HuggingFace
+Spaces as described in [our
+documentation](./docker.md#zenml-server-configuration-options). You should be
```
Comment (davanstrien) [inline docs/hub/spaces-sdks-docker-zenml.md:112]: https://docs.zenml.io/getting-started/deploying-zenml/docker#zenml-server-configuration-options?

**Review by davanstrien — CHANGES_REQUESTED** [review]
I added a few questions and suggestions.

File: docs/hub/spaces-sdks-docker-zenml.md (hunk 2 above)
Comment (strickvl) [reply docs/hub/spaces-sdks-docker-zenml.md]: Correct. Thanks for catching that!

File: docs/hub/spaces-sdks-docker-zenml.md (hunk 3 above)
Comment (strickvl) [reply docs/hub/spaces-sdks-docker-zenml.md]: Correct again. Fixed now.

**Comment by davanstrien** [comment]
Preview of the select spaces template view 
Screenshot 2023-03-27 at 15 11 42

I will update this on https://huggingface.co/datasets/huggingface/documentation-images

**Comment by davanstrien** [comment]
@osanseviero think this should be good to go now, but we might want to hold off merging until we're ready to launch.

**Comment by davanstrien** [comment]
@osanseviero you okay with me merging this now?

**Pull request merged**